add-chain/cmd/check-genesis.go

```diff
@@ -0,0 +1,54 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"os"
+
+	"github.com/ethereum-optimism/superchain-registry/add-chain/flags"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/urfave/cli/v2"
+)
+
+var CheckGenesisCmd = cli.Command{
+	Name:  "check-genesis",
+	Flags: []cli.Flag{flags.GenesisFlag, flags.ChainIdFlag},
+	Usage: "Sanity check genesis (genesis.json) is reproducible",
+	Action: func(ctx *cli.Context) error {
+		genesisPath := ctx.String(flags.GenesisFlag.Name)
+		fmt.Printf("Attempting to read from %s\n", genesisPath)
+		file, err := os.ReadFile(genesisPath)
+		if err != nil {
+			return fmt.Errorf("failed to read from local genesis.json config file: %w", err)
+		}
+		var localGenesis *core.Genesis
+		if err = json.Unmarshal(file, &localGenesis); err != nil {
+			return fmt.Errorf("failed to unmarshal local genesis.json into core.Genesis struct: %w", err)
+		}
+
+		chainId := ctx.Uint64(flags.ChainIdFlag.Name)
+		gethGenesis, err := core.LoadOPStackGenesis(chainId)
+		if err != nil {
+			return fmt.Errorf("failed to load genesis via op-geth: %w", err)
+		}
```

Comment:
Should we put a comment in, and/or a message if we get an error here, about requiring the chain to have been added already to the registry before running this command (in a separate `go run` invocation / build)?

Reply:
Added more detailed error message